### PR TITLE
feat: add elementwise select and with_columns to new streaming engine

### PR DESCRIPTION
**Pull request by orlp**

### CODE REVIEW

File: crates/polars-stream/src/nodes/in_memory_sink.rs [hunk 1]
```diff
@@ -51,8 +57,18 @@ impl ComputeNode for InMemorySink {
     }
 
     fn finalize(&mut self) -> PolarsResult<Option<DataFrame>> {
-        let mut morsels_per_pipe = core::mem::take(&mut *self.morsels_per_pipe.get_mut());
+        let morsels_per_pipe = core::mem::take(&mut *self.morsels_per_pipe.get_mut());
         let dataframes = linearize(morsels_per_pipe);
-        Ok(Some(accumulate_dataframes_vertical_unchecked(dataframes)))
+        if dataframes.is_empty() {
+            let cols = self
```
Comment (ritchie46) [inline crates/polars-stream/src/nodes/in_memory_sink.rs:63]: There is a `impl From<&Schema> for DataFrame`

File: crates/polars-stream/src/physical_plan/mod.rs [hunk 2]
```diff
@@ -25,16 +25,23 @@ pub enum PhysNode {
...
+        extend_original: bool,
+        schema: Arc<Schema>,
+    },
     Filter {
         input: PhysNodeKey,
         predicate: ExprIR,
     },
     SimpleProjection {
         input: PhysNodeKey,
-        schema: SchemaRef,
```
Comment (ritchie46) [inline crates/polars-stream/src/physical_plan/mod.rs:34]: Why not `SchemaRef`?  It's a bit easier to write. :)

File: crates/polars-stream/src/physical_plan/lower_ir.rs [hunk 3]
```diff
@@ -17,13 +17,50 @@ pub fn lower_ir(
...
+            let schema = schema.clone();
+            let input = lower_ir(*input, ir_arena, expr_arena, phys_sm)?;
+            Ok(phys_sm.insert(PhysNode::Select {
+                input,
+                selectors,
+                schema,
+                extend_original: true,
+            }))
+        },
+
```
Comment (ritchie46) [inline crates/polars-stream/src/physical_plan/lower_ir.rs:63]: Do we miss a `SimpleProjection` here?

File: crates/polars-stream/src/physical_plan/mod.rs (hunk 2 above)
Comment (orlp) [reply crates/polars-stream/src/physical_plan/mod.rs:34]: I think it's barely easier to write, but makes it harder to understand what's going on. I'd rather know directly that I have an `Arc` to a `Schema`.

File: crates/polars-stream/src/physical_plan/lower_ir.rs (hunk 3 above)
Comment (orlp) [reply crates/polars-stream/src/physical_plan/lower_ir.rs:63]: Yes, I can add it as an optimization in a follow-up PR.

File: crates/polars-stream/src/nodes/in_memory_sink.rs (hunk 1 above)
Comment (orlp) [reply crates/polars-stream/src/nodes/in_memory_sink.rs:63]: Not anymore :) It's now `DataFrame::empty_with_schema`, but am using it now.

File: crates/polars-stream/src/physical_plan/lower_ir.rs (hunk 3 above)
Comment (orlp) [reply crates/polars-stream/src/physical_plan/lower_ir.rs:63]: Oh I didn't realize it was already part of the IR, added it now.

**Pull request merged**